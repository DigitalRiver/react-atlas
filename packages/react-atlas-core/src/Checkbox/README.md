Checkbox:

    <Checkbox
	    label="Checkbox"
	  />

Checked:	  

      <Checkbox
        label="Checkbox Checked"
        checked
      />

Disabled:

      <Checkbox
        disabled
        label="Checkbox Disabled"
      />

Left label:

      <Checkbox
        label="Checkbox Left Label"
        labelPosition="left"
        checked
      />

Custom onClick method:

      {}

      <Checkbox
        label="Checkbox onClick"
        value="Bat"
        onClick={function(value, event, isValid, checked) { console.log(value, event, isValid, checked); }}
      />

onChange callback:

    <Checkbox
      label="Checkbox onChange"
      value="Duck"
      onChange={function(value, event, isValid, checked) { console.log(value, event, isValid, checked); }}
      />

No label:

      <Checkbox
      />

No label and checked:

      <Checkbox
        checked
      />

Required:

	  <Checkbox
	  	label="Required Example"
	  	required
	  />

Custom validation:

	  <Checkbox
	  	label="Required Example"
	  	errorCallback={ function(event, checked){ return {valid: checked, message: "Checkbox is required"} } }
	  />

Custom function before change:

      <Checkbox
      	label="Checkbox onBeforeChange"
      	onBeforeChange={ function(value){ if (!value) { let accept = confirm("Do you want to check this?"); return accept; } return true; } }
      />
      
Label and ID:	  

    <Checkbox
      label="Checkbox"
      id="my-checkbox"
    />
