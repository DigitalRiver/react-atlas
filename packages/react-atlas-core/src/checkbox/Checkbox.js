import React, { Component, PropTypes } from 'react';
import themeable from 'react-themeable';
import { classNames } from '../utils/utils';


/**
 * Simple component for a basic checkbox
 */
const Checkbox = ({title, label, disabled, inline, ...other}) => {
      //const cx = ClassNames.bind(other.theme);
      const theme = themeable(other.theme);
      const componentClasses = classNames({
          "block": !inline,
          inline,
          disabled,
          [className]: className
      });

      title = title ? title : label;

      return (
          <label {...theme(1, ...componentClasses)} title={title}>
              <input {...other} type="checkbox" disabled={disabled} {...theme(2, 'input')} />
              {label && <span {...theme(3, 'label')}>{label}</span>}
          </label>
      );
};

Checkbox.propTypes = {
  /**
   * Text for checkbox label
   * @examples 'Some Label'
   */
  label: PropTypes.string,
	/**
   * A css class name that will be appended to the wrapping <label> element around the <input> and <span> elements.
   */
  className: PropTypes.string,
	/**
   * If included, checkbox is disabled
   * @examples <Checkbox disabled />, <Checkbox disabled={true} />
   */
  disabled: PropTypes.bool,
	/**
   * If included, renders the checkbox and it's label inline, so it can be side-by-side to other content.
   */
  inline: PropTypes.bool,
  /**
   * Text for checkbox label title. (i.e. "alt-text" for checkboxes, useful for accessibility). If not provided, will be label text.
   * @examples 'Some Title'
   */
  title: PropTypes.string,
	/**
   * Defines if checkbox should be checked on load.
   */
  defaultChecked: PropTypes.bool
};

Checkbox.defaultProps = {
  className: '',
  disabled: false,
  inline: false
};

Checkbox.styleguide = {
  category: "Form Components",
  index: "3.2",
  example: `
<section style={{textAlign:"left"}}>
  <h4 style={{marginBottom: "10px"}}>Checkboxes</h4>
  <Checkbox
    defaultChecked={true}
    label="Checked"
  />
  <Checkbox
    defaultChecked={false}
    label="Not checked"

  />
  <Checkbox
    defaultChecked={true}
    label="Disabled checkbox"
    disabled
  />
  <Checkbox
    defaultChecked={false}
    label="Inline checkbox 1"
    inline
  />
  <Checkbox
    defaultChecked={false}
    label="Inline checkbox 2"
    inline
  />
</section>
`
};

export default Checkbox;
