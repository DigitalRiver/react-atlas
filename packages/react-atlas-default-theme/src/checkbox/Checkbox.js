import { Checkbox } from "react-atlas-core";
import styles from "./checkbox.css";
import createComponent from "../createComponent";

export default createComponent(Checkbox, styles);
