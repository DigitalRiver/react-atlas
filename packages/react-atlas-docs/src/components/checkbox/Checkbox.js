import React, { PropTypes } from "react";
import * as Atlas from 'react-atlas';

export default class Checkbox extends React.Component {
  constructor(props){
    super(props)
  }
  render() {
    return (
      <Atlas.Checkbox {...this.props}></Atlas.Checkbox>
    )
  }
}