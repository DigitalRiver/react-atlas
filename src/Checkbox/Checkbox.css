@import '../styles/variables.css';

.disabled {
    composes: default-text from '../styles.css';
}

.relative {
    position:relative;
}

.padding {
    padding-bottom: 5px;
}

.input {
    composes: relative;
    top: 1px;
}

.checkbox {
  composes: rounded from '../styles.css';
  composes: default-border from "../styles.css";
  composes: relative;
  position: absolute;
  top: 0px;
  width: 16px;
  height: 16px;
  cursor: pointer;
}

.disabled {
  opacity: 0.65;
  composes: cursor-not-allowed from '../styles.css'; 
  pointer-events: none;
}

.checked {
  composes: checkbox;
  composes: bg-primary from '../styles.css';
}

.hidden {
  display: none;
  visibility: hidden;
}

.focus {
  box-shadow: 0 0 6px var(--primary)
}

.checkmark {
  composes: relative;
  top: 0px;
  left: 4px;
  width: 4px;
  height: 10px;
  transform: rotate(45deg);
  border: solid var(--white);
  border-width: 0 3px 3px 0;    
}

.error {
  border: solid 1px var(--error);
}

/* Display checkbox with out the checkbox selected.  */
.not_checked {
  composes: checkbox;
  background: -webkit-linear-gradient(#fff, #e6e6e6); /* For Safari 5.1 to 6.0 */
  background: -o-linear-gradient(#fff, #e6e6e6); /* For Opera 11.1 to 12.0 */
  background: -moz-linear-gradient(#fff, #e6e6e6); /* For Firefox 3.6 to 15 */
  background: linear-gradient(#fff, #e6e6e6); /* Standard syntax */
}

.label {
  composes: default-font from '../styles.css';
  cursor: pointer;
  white-space: nowrap;
  padding-right:5px;
}

/* label text */
.label_left {
  composes: label;
  padding-right: 2px;
  margin-left: 0px;
}

.error_text {
  composes: default-font from '../styles.css';
  composes: error-text from '../styles.css';
}

.error_message {
  composes: error_text;
  margin-left:23px;
}

.inline_block {
  display: inline-block;
}

.checkbox_padding {
  padding: 10px 10px 0 0;
}

.inline_padding {
  padding: 2px 10px 0 0;
}

.float_left {
  float:left;
  padding-right: 3px;
}

.float_right {
  float:right;
  padding-left: 3px;
  position: relative;
}

.opacity {
  /*this needs to stay here to avoid display issues with those modules in IE*/
  opacity: 0;
}

.required {
  composes: secondary-text from '../styles.css';
}

.required_error {
  composes: error-text from '../styles.css';
}