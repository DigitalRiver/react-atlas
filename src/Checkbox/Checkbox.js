import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import { utils } from "../utils";
import messages from "../utils/messages.js";
import CSSModules from "react-css-modules";
import styles from "./Checkbox.css";
import { Text } from "../Text";
import blacklist from "blacklist";

/**
 * Simple component for a basic checkbox
 */
export class Checkbox extends React.PureComponent {
  constructor(props) {
    super(props);

    // Initial state
    this.state = {
      "checked": this.props.checked || false,
      "status": props.status || null,
      "message": props.message || null,
      "focus": false
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (nextProps.checked !== this.props.checked) {
      this.setState({
        "checked": nextProps.checked || false
      });
    }
  }
  // Handles new checkbox clicks and sets value and checked status of hidden input
  _clickHandler = event => {
    if (this.props.disabled) {
      return false;
    }
    event.persist();
    if (typeof this.props.onBeforeChange !== "undefined") {
      let result = this.props.onBeforeChange(this.state.checked);
      if (result === false) {
        return;
      }
    }
    const validationObject = this._validationHandler(this.props.valid);
    this.setState(
      {
        "checked": !this.state.checked,
        "status": validationObject.status,
        "message": validationObject.message
      },
      function() {
        const data = {
          "value": this.props.value,
          "status": this.state.status,
          "checked": this.state.checked
        };
        /* Check if onClick has been passed, if so call it. */
        if (typeof this.props.onClick !== "undefined") {
          this.props.onClick(data);
        }
        /* Check if onChange has been passed, if so call it. */
        if (typeof this.props.onChange !== "undefined") {
          this.props.onChange(data);
        }
      }
    );
  };

  // Added to remove console warning for controlled/uncontrolled component
  _handleChange = () => {
    return false;
  };

  _getStatus = preState => {
    let status = "error";
    if (
      preState &&
      (this.props.required && !this.state.checked || !this.props.required)
    ) {
      status = null;
    } else if (
      !preState &&
      (this.props.required && this.state.checked || !this.props.required)
    ) {
      status = null;
    }
    return status;
  };

  _validationHandler = callback => {
    // If custom validation callback is provided set validationObject with response, otherwise check if required
    const validationObject = callback
      ? callback(event, !this.state.checked)
      : {
          "status": this._getStatus(true),
          "message": messages.requiredMessage
        };
    return validationObject;
  };

  // Used from a parent Form component to validate only
  _formValidate = () => {
    const validationObject = this.props.valid
      ? this.props.valid(event, !this.state.checked)
      : {
          "status": this._getStatus(false),
          "message": messages.requiredMessage
        };
    this.setState({
      "status": validationObject.status,
      "message": validationObject.message
    });
    return validationObject;
  };

  render() {
    const {
      className,
      disabled,
      groupError,
      hidden,
      id,
      inline,
      label,
      labelPosition,
      required,
      status,
      style,
      title,
      ...others
    } = this.props;

    // Declaring the following variables so they don't get passed to the input through the prop spread.
    const othersFiltered = blacklist(others, "onBeforeChange", "valid");

    const inlineCheckbox = cx({
      "inline_block": inline,
      "checkbox_padding": !inline,
      "inline_padding": inline,
      hidden
    });
    const labelStyle = cx({
      "label": labelPosition !== "left",
      "label_left": labelPosition === "left"
    });
    const checkboxDisplay =
      labelPosition === "left" ? "float_right" : "float_left";
    const title_label = title ? title : label;
    let disabledClass = cx({
      disabled,
      "inline_block": true,
      "relative": true,
      "padding": !inline
    });
    const error = groupError || this.state.status === "error";
    let checkboxClass = cx({
      "checked": this.state.checked,
      "error": error,
      "not_checked": !this.state.checked,
      "focus": this.state.focus
    });

    // Gets the appropriate jsx to render a "required" identifier next to the Checkbox.
    const reqText = typeof required === "string" ? required : "*";

    // Gets the appropriate jsx to render an error message below the Checkbox.
    const errorMessage =
      this.state.status === "error"
        ? utils.getErrorMessage(this.state.message)
        : null;

    const requiredClasses = cx({
      "required": true,
      "required_error": status === "error"
    });

    return (
      <div
        onClick={this._clickHandler}
        styleName={inlineCheckbox}
        style={style}
        className={cx(className)}
      >
        <div styleName={disabledClass}>
          {label && 
            <label styleName={labelStyle} title={title_label} htmlFor={id}>
              {label}
            </label>
          }
          <div styleName={checkboxDisplay}>
            <input
              {...othersFiltered}
              label={label}
              type="checkbox"
              disabled={disabled}
              checked={this.state.checked}
              required={required}
              hidden={hidden}
              id={id}
              onChange={this._handleChange}
              styleName={cx("opacity")}
            />
            <div styleName={checkboxClass}>
              {this.state.checked && <div styleName={"checkmark"} />}
            </div>
          </div>
          {required &&
            required !== "" && 
              <span styleName={requiredClasses}>
                {" "}
                <Text>{reqText}</Text>
              </span>
            }
        </div>
        {errorMessage}
      </div>
    );
  }
}

Checkbox.propTypes = {
  /**
   * When true, Checkbox will be checked on load.
   */
  "checked": PropTypes.bool,

  /** An object, array, or string of CSS classes to apply to Checkbox.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * If included, checkbox is disabled
   * @examples <Checkbox disabled />, <Checkbox disabled={true} />
   */
  "disabled": PropTypes.bool,

  /**
   * For use when an error state has been passed down from the parent CheckboxGroup.
   */
  "groupError": PropTypes.bool,
  /**
   * Determines if the Checkbox is hidden.
   * @examples '<Checkbox hidden/>'
   */
  "hidden": PropTypes.bool,

  /** Will set the html "id" property on the Checkbox.*/
  "id": PropTypes.string,

  /**
   * When true, will render the Checkbox and its label inline.
   */
  "inline": PropTypes.bool,

  /**
   * Text for Checkbox label.
   * @examples 'Some Label'
   */
  "label": PropTypes.string,

  /**
   * Will position the Checkbox label to the left or the right of the checkbox.
   */
  "labelPosition": PropTypes.string,

  /**
   * Function that will be executed before changing the "checked" state of the Checkbox.
   */
  "onBeforeChange": PropTypes.func,

  /**
   * Function that will be executed when the Checkbox state is changed.
   */
  "onChange": PropTypes.func,

  /**
   * Function that will be executed on click.
   */
  "onClick": PropTypes.func,

  /**
   * Sets the Checkbox as required. Will be validated in _validationHandler. Accepts a boolean or a string.
   * If a string is passed it will be displayed instead of the traditional * next to the Checkbox.
   * @examples '<TextField required/>'
   */
  "required": PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object,

  /**
   * Text for Checkbox label title. (i.e. "alt-text" for checkboxes, useful for accessibility). If not provided, label text will be used.
   * @examples 'Some Title'
   */
  "title": PropTypes.string,

  /**
   * Used to pass a function for custom validation. Should return either true or false.
   */
  "valid": PropTypes.func,

  /**
   * The value of the Checkbox.
   */
  "value": PropTypes.string,

  /**
   * Sets the status of the Checkbox. Options are null, "success", "error", and "warning".
   * @examples '<Checkbox status="error" />'
   */
  "status": PropTypes.string,

  /**
   * Sets the status message of the Checkbox.
   * @examples '<Checkbox
   label="Required Example"
   valid={ function(checked){ return {status: "success", message: "Checkbox is required"} } }
   />'
   */
  "message": PropTypes.string
};

Checkbox.defaultProps = {
  "className": "",
  "disabled": false,
  "inline": false,
  "checked": false,
  "required": false
};

export default CSSModules(Checkbox, styles, { "allowMultiple": true });
