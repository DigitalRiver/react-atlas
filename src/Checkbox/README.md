Checkbox:

    <Checkbox
	    label="Checkbox"
	  />

Checked:	  

      <Checkbox
        label="Checkbox Checked"
        checked
      />

Disabled:

      <Checkbox
        disabled
        label="Checkbox Disabled"
      />

Left label:

      <Checkbox
        label="Checkbox Left Label"
        labelPosition="left"
        checked
      />

Custom onClick method:

      {}

      <Checkbox
        label="Checkbox onClick"
        value="Bat"
        onClick={function(data) { console.log(data.value, data.status, data.checked); }}
      />

onChange callback:

    <Checkbox
      label="Checkbox onChange"
      value="Duck"
      onChange={function(data) { console.log(data.value, data.status, data.checked); }}
      />

No label:

      <Checkbox
      />

No label and checked:

      <Checkbox
        checked
      />

Required:

	  <Checkbox
	  	label="Required Example"
	  	required
	  />

Custom validation:

	  <Checkbox
	  	label="Required Example"
	  	required="Custom Validation Example"
	  	valid={ function(event, checked){ return {status: checked ? null : "error", message: "Checkbox is required"} } }
	  />

Custom function before change:

      <Checkbox
      	label="Checkbox onBeforeChange"
      	onBeforeChange={ function(value){ if (!value) { let accept = confirm("Do you want to check this?"); return accept; } return true; } }
      />
      
Label and ID:	  

    <Checkbox
      label="Checkbox"
      id="my-checkbox"
    />
