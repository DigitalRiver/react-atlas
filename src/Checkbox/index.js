import Checkbox from "./Checkbox.js";
export { Checkbox };
