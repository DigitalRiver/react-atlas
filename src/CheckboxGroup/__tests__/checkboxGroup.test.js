import React from "react";
import { mount } from "enzyme";
import { CheckboxGroup } from "../index";
import { Checkbox } from "../../Checkbox/index";

import renderer from "react-test-renderer";

describe("Test CheckboxGroup component - the basics", () => {
  it("Checkbox group - 4 checkboxes, 1 checked", function() {
    const component = mount(
      <CheckboxGroup title={"CheckboxGroup test"}>
        <Checkbox name={"1"} />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
      </CheckboxGroup>
    );
    component
      .childAt(0)
      .childAt(1)
      .simulate("click");
    component.update();
    expect(component.state().totalChecked).toBe(1);
  });

  it("Checkbox group - 4 checkboxes, 1 checked (with onChange)", function() {
    const component = mount(
      <CheckboxGroup
        title={"CheckboxGroup test"}
        onChange={() => {
          console.log("onChange triggered!!");
        }}
      >
        <Checkbox name={"1"} />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
      </CheckboxGroup>
    );

    component
      .childAt(0)
      .childAt(1)
      .simulate("click");
    component.update();
    expect(component.state().totalChecked).toBe(1);
  });

  it("Test renders correctly", () => {
    const comp = 
      <CheckboxGroup title={"CheckboxGroup test"}>
        <Checkbox name={"1"} />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
      </CheckboxGroup>
    ;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Test CheckboxGroup component - the basics", () => {
  it("Checkbox group - 4 checkboxes, 1 checked", function() {
    const component = mount(
      <CheckboxGroup title={"CheckboxGroup test"}>
        <Checkbox name={"1"} />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
      </CheckboxGroup>
    );

    component
      .childAt(0)
      .childAt(1)
      .simulate("click");
    component.update();
    expect(component.state().totalChecked).toBe(1);
  });

  it("Checkbox group - 4 checkboxes, 2 checked 1 from creation", function() {
    const component = mount(
      <CheckboxGroup title={"CheckboxGroup test"}>
        <Checkbox name={"1"} checked />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
      </CheckboxGroup>
    );

    component
      .childAt(0)
      .childAt(2)
      .simulate("click");
    component.update();
    expect(component.state().totalChecked).toBe(2);
  });

  it("Checkbox group - 4 checkboxes, 0 checked with 1 checked from creation", function() {
    const component = mount(
      <CheckboxGroup title={"CheckboxGroup test"}>
        <Checkbox name={"1"} />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
      </CheckboxGroup>
    );

    component
      .childAt(0)
      .childAt(0)
      .simulate("click");

    expect(component.state().totalChecked).toBe(0);
  });

  it("Checkbox group - 4 checkboxes, 3 checked", function() {
    const component = mount(
      <CheckboxGroup title={"CheckboxGroup test"}>
        <Checkbox name={"1"} />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
      </CheckboxGroup>
    );

    component
      .childAt(0)
      .childAt(1)
      .simulate("click");
    component
      .childAt(0)
      .childAt(2)
      .simulate("click");
    component
      .childAt(0)
      .childAt(3)
      .simulate("click");
    component.update();
    expect(component.state().totalChecked).toBe(3);
  });

  it("Checkbox group - 4 checkboxes, 2 checked", function() {
    const component = mount(
      <CheckboxGroup title={"CheckboxGroup test"}>
        <Checkbox name={"1"} />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
      </CheckboxGroup>
    );

    component
      .childAt(0)
      .childAt(1)
      .simulate("click");
    component
      .childAt(0)
      .childAt(2)
      .simulate("click");
    component
      .childAt(0)
      .childAt(3)
      .simulate("click");
    component.update();
    expect(component.state().totalChecked).toBe(3);

    component
      .childAt(0)
      .childAt(1)
      .simulate("click");
    component.update();
    expect(component.state().totalChecked).toBe(2);
  });
});

function _verifyMinMax(mi, ma, numberOfClicks, message, expectedResult) {
  const component = mount(
    <CheckboxGroup title={"CheckboxGroup test"} min={mi} max={ma}>
      <Checkbox name={"1"} />
      <Checkbox name={"2"} />
      <Checkbox name={"3"} />
      <Checkbox name={"4"} />
      <Checkbox name={"5"} />
    </CheckboxGroup>
  );
  if (message !== "") {
    component.limitMessage = message;
  }
  for (let i = 1; i <= numberOfClicks; i++) {
    component
      .childAt(0)
      .childAt(i)
      .simulate("click");
  }
  component.update();
  expect(component.state().groupError).toBe(expectedResult);
}

describe("Test CheckboxGroup component - Min/Max tests", () => {
  it("Checkbox group - 5 checkboxes, max=4, min=2, 1 checked", function() {
    _verifyMinMax(2, 4, 1, "Limit exceed!!", true);
  });
  it("Checkbox group - 5 checkboxes, max=4, min=2, 2 checked", function() {
    _verifyMinMax(2, 4, 2, "Limit exceed!!", false);
  });
  it("Checkbox group - 5 checkboxes, max=4, min=2, 3 checked", function() {
    _verifyMinMax(2, 4, 3, "Limit exceed!!", false);
  });
  it("Checkbox group - 5 checkboxes, max=4, min=2, 4 checked", function() {
    _verifyMinMax(2, 4, 4, "Limit exceed!!", false);
  });
  it("Checkbox group - 5 checkboxes, max=4, min=2, 5 checked", function() {
    _verifyMinMax(2, 4, 5, "Limit exceed!!", true);
  });
  it("Checkbox group - 5 checkboxes, max=4, min=2, 1 checked no Limmit message", function() {
    _verifyMinMax(2, 4, 1, "", true);
  });
  it("Checkbox group - 5 checkboxes, max=4, min=2, 5 checked no Limmit message", function() {
    _verifyMinMax(2, 4, 5, "", true);
  });

  it("Checkbox group - 5 checkboxes, max=4, min=2, 3 checked", function() {
    const component = mount(
      <CheckboxGroup
        title={"CheckboxGroup test"}
        max={4}
        min={2}
        limitMessage={"Limit exceed!!"}
      >
        <Checkbox name={"1"} />
        <Checkbox name={"2"} />
        <Checkbox name={"3"} />
        <Checkbox name={"4"} />
        <Checkbox name={"5"} />
      </CheckboxGroup>
    );

    for (let i = 1; i <= 5; i++) {
      component
        .childAt(0)
        .childAt(i)
        .simulate("click");
    }
    component.update();
    expect(component.state().groupError).toBe(true);

    component
      .childAt(0)
      .childAt(2)
      .simulate("click");
    component
      .childAt(0)
      .childAt(0)
      .simulate("click");
    component.update();
    expect(component.state().groupError).toBe(false);
  });
});
