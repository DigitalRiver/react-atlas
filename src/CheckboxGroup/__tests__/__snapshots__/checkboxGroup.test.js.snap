// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckboxGroup component - the basics Test renders correctly 1`] = `
<div
  className="checkboxGroup"
  style={undefined}
>
  <div>
    <div
      className="tooltipRight labelContainer"
    >
      <div
        className="verticalPadding horizontalPadding"
      >
        <label
          className="labelSpacing label label"
          htmlFor={undefined}
          id={undefined}
          style={undefined}
        >
          CheckboxGroup test
        </label>
      </div>
    </div>
  </div>
  <div
    className="checkbox_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="opacity"
          disabled={false}
          hidden={undefined}
          id={undefined}
          label={undefined}
          name="1"
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <div
          className="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    className="checkbox_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="opacity"
          disabled={false}
          hidden={undefined}
          id={undefined}
          label={undefined}
          name="2"
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <div
          className="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    className="checkbox_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="opacity"
          disabled={false}
          hidden={undefined}
          id={undefined}
          label={undefined}
          name="3"
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <div
          className="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    className="checkbox_padding"
    onClick={[Function]}
    style={undefined}
  >
    <div
      className="inline_block relative padding"
    >
      <div
        className="float_left"
      >
        <input
          checked={false}
          className="opacity"
          disabled={false}
          hidden={undefined}
          id={undefined}
          label={undefined}
          name="4"
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <div
          className="not_checked"
        />
      </div>
    </div>
  </div>
</div>
`;
